### PR TITLE
Add flannel adr

### PR DESCRIPTION
**Pull request by dereknola**
Signed-off-by: Derek Nola <derek.nola@suse.com>

#### Proposed Changes ####
- Adds the flannel ADR to set the groundwork for discussion on the https://github.com/k3s-io/k3s/pull/6897

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: docs/adrs/flannel-options.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+* Use it for potential new flannel toggles
+
+## Proposal
+
+Starting in v1.26, introduce a new `flannel-opts` flag that includes flannel server options. The redundant flags are deprecated and removed in a few releases.
+
+We could reduce it to 2:
+
+* flannel-backend (string)
+* flannel-opts ([]string)
```
Comment (brandond) [inline docs/adrs/flannel-options.md:92]: +1 on leaving flannel-backend around, but moving everything else into flannel-opt

File: docs/adrs/flannel-options.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+Flannel-opts would have the following values:
```
Comment (brandond) [inline docs/adrs/flannel-options.md:94]: I would suggest making this `--flannel-opt`, but accepting either multiple comma-separated values, or being repeated multiple times with the values all being collected. This follow the convention of other flags such as `--kubelet-arg` - the flag name is singular (opt, arg, and so on).

Comment (dereknola) [reply docs/adrs/flannel-options.md]: On the naming, that was my thought as well. I have included your thought on being able to repeat the arg multiple times.

File: docs/adrs/flannel-options.md [hunk 3]
```diff
@@ -84,25 +84,32 @@ The suggestion is to include a string slice flag that would consolidate all flan
...
 | `external-ip` | `--flannel-external-ip` |
 
+Similar to how we handle `--kubelet-arg`, both comma separated lists and repeated args would be accepted.
+
 Examples of usage:  
 `--flannel-opts=ipv6-masq,external-ip` (assumes true)  
 `--flannel-opts=ipv6-masq=true`  
 `--flannel-opts=ipv6-masq=true,external-ip=false`  
+`--flannel-opts=ipv6-masq --flannel-opts=external-ip`
+`--flannel-opts=ipv6-masq --flannel-opts=external-ip=false`
```
Comment (brandond) [inline docs/adrs/flannel-options.md:108]: flannel-opt for all of these

File: docs/adrs/flannel-options.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+** Does not provide much value as flannel in k3s can configure the interface per node
+
+Regarding subnet and backend configuration:
+
+`Network` and `IPv6Network` are set by reading the assigned podCIDR for the Kubernetes node.
+`EnableIPv6` is set based on what the user passed as k3s server `--cluster-cidr`.
+`Backend` uses vxlan as default, although this can be changed using k3s server flag `--flannel-backend`
+
+Something important to note is that k3s allows the user to override the whole subnet and backend configuration, by using the k3s agent flag `--flannel-conf` and even the flannel CNI configuration file, by using the k3s agent flag `--flanel-cni-conf`. Overriding the whole subnet and backend configuration is useful if user wants to add specific backend configurations, e.g. VNI for vxlan. Users can currently do this by using `<=option1=val1,option2=val2>` when selecting the backend but it seldom used and complicates things, thus it was deprecated in favor of `--flannel-cni-conf`.
```
[the remark this answers is not in the document]
Comment (dereknola) [reply docs/adrs/flannel-options.md:63]: I'm not super clear on the difference between `--flannel-conf` and `--flannel-cni-conf`, I pulled that info from Manuel's original ADR. I am not sure how easy it is to add a <key, value> system when the config has nested components. I was trying to contain `flannel-opts` to only booleans.

[the remark this answers is not in the document]
Comment (rbrtbnfgl) [reply docs/adrs/flannel-options.md:63]: `--flannel-cni-conf` is the configuration file used by kubelet to configure the network when a new pod is created.
`--flannel-conf` is the flannel CNI configuration that contains the clusterCIDR and the backend type and configurations.

[the remark this answers is not in the document]
Comment (dereknola) [reply docs/adrs/flannel-options.md]: I have incorporated your info into the ADR, thanks!

File: docs/adrs/flannel-options.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+** Does not provide much value as flannel in k3s can configure the interface per node
+
+Regarding subnet and backend configuration:
+
+`Network` and `IPv6Network` are set by reading the assigned podCIDR for the Kubernetes node.
+`EnableIPv6` is set based on what the user passed as k3s server `--cluster-cidr`.
+`Backend` uses vxlan as default, although this can be changed using k3s server flag `--flannel-backend`
+
+Something important to note is that k3s allows the user to override the whole subnet and backend configuration, by using the k3s agent flag `--flannel-conf` and `--flanel-cni-conf`. Overriding the whole subnet and backend configuration is useful if user wants to add specific backend configurations, e.g. VNI for vxlan. Users can currently do this by using `<=option1=val1,option2=val2>` when selecting the backend, but it seldom used and complicates things, thus it was deprecated in favor of `--flannel-cni-conf`.
```
Comment (rbrtbnfgl) [inline docs/adrs/flannel-options.md:63]:
```suggestion
Something important to note is that k3s allows the user to override the whole subnet and backend configuration, by using the k3s agent flag `--flannel-conf` and `--flanel-cni-conf`. Overriding the whole subnet and backend configuration is useful if user wants to add specific backend configurations, e.g. VNI for vxlan. Users can currently do this by using `<=option1=val1,option2=val2>` when selecting the backend, but it seldom used and complicates things, thus it was deprecated in favor of `--flannel-conf`.
```

**Review by briandowns — APPROVED** [review]

**Review by cwayne18 — APPROVED** [review]

**Review by rbrtbnfgl — APPROVED** [review]

**Pull request merged**